pkg/config/remote/service/service.go

```diff
@@ -511,45 +534,74 @@ func (s *CoreAgentService) Start() {
 		defer func() {
 			close(s.stopOrgPoller)
 		}()
+		if s.disableConfigPollLoop {
+			startWithoutAgentPollLoop(s)
+		} else {
+			startWithAgentPollLoop(s)
+		}
 
-		err := s.refresh()
-		if err != nil {
-			if s.previousOrgStatus != nil && s.previousOrgStatus.Enabled && s.previousOrgStatus.Authorized {
-				log.Errorf("[%s] Could not refresh Remote Config: %v", s.rcType, err)
-			} else {
-				log.Debugf("[%s] Could not refresh Remote Config (org is disabled or key is not authorized): %v", s.rcType, err)
-			}
+	}()
+}
+
+func startWithAgentPollLoop(s *CoreAgentService) {
+	err := s.refresh()
+	if err != nil {
+		if s.previousOrgStatus != nil && s.previousOrgStatus.Enabled && s.previousOrgStatus.Authorized {
+			log.Errorf("[%s] Could not refresh Remote Config: %v", s.rcType, err)
+		} else {
+			log.Debugf("[%s] Could not refresh Remote Config (org is disabled or key is not authorized): %v", s.rcType, err)
 		}
+	}
 
-		for {
-			var err error
-			refreshInterval := s.calculateRefreshInterval()
-			select {
-			case <-s.clock.After(refreshInterval):
+	for {
+		var err error
+		refreshInterval := s.calculateRefreshInterval()
+		select {
+		case <-s.clock.After(refreshInterval):
+			err = s.refresh()
+		// New clients detected, request refresh
+		case response := <-s.cacheBypassClients.requests:
+			if !s.cacheBypassClients.Limit() {
 				err = s.refresh()
-			// New clients detected, request refresh
-			case response := <-s.cacheBypassClients.requests:
-				if !s.cacheBypassClients.Limit() {
-					err = s.refresh()
-				} else {
-					s.telemetryReporter.IncRateLimit()
-				}
-				close(response)
-			case <-s.stopConfigPoller:
-				log.Infof("[%s] Stopping Remote Config configuration poller", s.rcType)
-				return
+			} else {
+				s.telemetryReporter.IncRateLimit()
 			}
+			close(response)
+		case <-s.stopConfigPoller:
+			log.Infof("[%s] Stopping Remote Config configuration poller", s.rcType)
+			return
+		}
 
-			if err != nil {
-				if s.previousOrgStatus != nil && s.previousOrgStatus.Enabled && s.previousOrgStatus.Authorized {
-					exportedLastUpdateErr.Set(err.Error())
-					log.Errorf("[%s] Could not refresh Remote Config: %v", s.rcType, err)
-				} else {
-					log.Debugf("[%s] Could not refresh Remote Config (org is disabled or key is not authorized): %v", s.rcType, err)
-				}
-			}
+		if err != nil {
+			logRefreshError(s, err)
 		}
-	}()
+	}
+}
+
+func startWithoutAgentPollLoop(s *CoreAgentService) {
+	for {
+		var err error
+		response := <-s.cacheBypassClients.requests
+		if s.forceCacheBypass || !s.cacheBypassClients.Limit() {
```

Comment:
I'm a little weary of bypassing the limiter outright. I think we need to evaluate what a sane max is before we start to actually limit the number of new clients, and set the max to that, after which any new tracers will not receive configurations

Reply:
That makes sense, updated per our conversation. It'll now bypass the cache if the limit hasn't been reached, otherwise it'll error out.